### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by samwel-mongare**
### The current branch has the following features to be merged into the development branch:

- A column species of type string in the animals table.
- Updated animal data
- Several created and completed transactions
- Inside a transaction functions to:

    - Delete all animals born after Jan 1st, 2022.
    - Create a savepoint for the transaction.
    - Update all animals' weight to be their weight multiplied by -1.
    - Commit transaction

- queries to answer the following questions:

    - How many animals are there?
    - How many animals have never tried to escape?
    - What is the average weight of animals?
    - Who escapes the most, neutered or not neutered animals?
    - What is the minimum and maximum weight of each type of animal?
    What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -11,3 +11,21 @@ VALUES('Pikachu','2021-01-07', 1 , false, 15.04);
 
 INSERT INTO animals (name,date_of_birth, escape_attempts, neutered, weight_kg)
 VALUES('Devimon','2017-05-12', 5 , true, 11.00);
+
+INSERT INTO animals (name,date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES('Charmander','2020-02-08', 0 , false, 11.00);
+
```
Comment (Meltrust) [inline data.sql:17]: - [ ] In squema.sql, you should add a new column called species with a string type, as per the project requirements:

image

Kindly add the species column to the animals table.

File: queries.sql [hunk 2]
```diff
@@ -29,4 +29,108 @@ SELECT * from animals WHERE name != 'Gabumon';
...
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg
 
-SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- TRANSACTIONS
+-- FIRST TRANSACTION
+
+BEGIN TRANSACTION;
```
Comment (Meltrust) [inline queries.sql:37]: According to the project requirements:

image

Therefore:

- [x] Please, verify each of your changes.  Verify means a SELECT statement.

Example:

```SQL
BEGIN; -- start a transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
```

Please, verify your data in this way every time the requirements ask for it.

That way, you can check how the data behaves 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Mosams,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice code organization✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (samwel-mongare) [reply data.sql:17]: Hello, as per the requirements, I have add the species column which an ALTER TABLE statement as it is the only way to add a new column to an existing table. I have added the same SQL statement in the schema.sql file. I will however include it in the previously created statement though I felt maintaining the project flow would have been important for future reference. Thank you for the suggestion

**Review by AshakaE — APPROVED** [review]
### Hi @Mosams   🙋‍♀️, 

# STATUS APPROVED ✔
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**